squad/api/views.py

```diff
@@ -92,7 +100,8 @@ def add_test_run(request, group_slug, project_slug, version, environment_slug):
     receive = ReceiveTestRun(project)
 
     try:
-        receive(**test_run_data)
+        testrun = receive(**test_run_data)
+        log_addition(request, testrun, "Test Run created")
```

Comment:
I figure you can't move these to core/tasks/__init__ because there's no request available there (it'd be nice because of the code repetition though). 
But can't you return a tuple of (testrun, build) if the build was created in ReceiveTestRun? It would be sensible afais.

Reply:
maybe this is an option. Let me take a look

Reply:
I fixed that.